src/Logging/HTTPOutputHandler.php

```diff
@@ -137,6 +138,16 @@ public function setDefaultFormatter(FormatterInterface $formatter)
         return $this;
     }
 
+    protected function shouldShowError(int $errorCode): bool
```

Comment:
```suggestion
    private function shouldShowError(int $errorCode): bool
```

Reply:
This is intentionally protected so that if someone wants to change what determines if an error should show for _their_ project, they can do so easily.

Reply:
So that implying they would subclass this, override the method and use Injector to use their the subclass instead?  That's not a great way to do things.

Should add an extension hook here if you think it's something projects would actually want ... however I'd say don't bother because we already have provided environment variables support. 

What the actual use case here? Project want to also not show E_USER_DEPRECATED from symony like dependencies in the browser but they want to log them?

Reply:
> So that implying they would subclass this, override the method and use Injector to use their the subclass instead? That's not a great way to do things.

That's literally what the injector is for. It's why we have config setting this class through the injector in the first place. An extension hook for this seems like overkill - it's not something that's going to be done frequently.

Reply:
OK sounds like there's some real-world uses for this.  Though I'd say just making a method protected so it can be  subclassed and overridden is philosophically the wrong approach to solving configurability:
- Making methods protected instead of private makes them API which increases the maintenance burden on the CMS side, the default should always be private and protected when there's an immediate reason.
- Project with files such as MyHTTPOutputHandler extends HTTPOutputHandler is just messy.  Sure this is a common way to work-around core bugs, but this seems like a poor way to handle configuration.
- We can solve this with an extension hook (maximum flexibility), or
- For a more defined API we'd use `private static config` variables.  We can also provide Environment variables to be extra nice like we've in the Deprecation class
- Though this is a fairly niche use case and it may be too rigid, so the extension hook seems like the best solution here

Reply:
This sounds like a wider discussion about our approach to configurability and extensibility in general and probably needs weigh in from @silverstripe/core-team

Reply:
I'm 50/50 on this. Extension points were the more common way in the past, but I get that we don't want to always stick with things from the past for everything. And having a class that extends this replaced by the injector is the same number of new classes as creating an extension and applying it, so no saving there.
There might be a bit of sentiment that injector is more magical than extensions and people are more familiar with extensions, in this case, if someone needs to change this they are probably already quite deep and would know how to replace a class using injector, so no deciding factor there either.
I also get the point of public API and maintainability vs niche use for this particular thing, so... ugh, no right or wrong here I guess. But I would proooobably lean towards private method with an extension point; old school. Let someone else also weigh in :-D

It's definitely a wider and longer-term discussion on the overall direction we wanna go in. Also, does an extension hook constitute a pseudo-public API anyway? Like if it's within a private method and we remove it later and switch to protected later, is it BC breaking?

Reply:
> does an extension hook constitute a pseudo-public API anyway?

Yup. So we're adding in some extra API that needs to accept a value and "return" (where an extension just changes the value, rather than returning it) a value. So no deciding factor from a maintenance burden pov either I don't think.

> Like if it's within a private method and we remove it later and switch to protected later, is it BC breaking?

If an extension hook is removed, it's breaking. If the extension hook is kept and added in the protected method in this scenario then it's not breaking (the same extension is still doing the same job)

Reply:
We've spoken about this at length in the past: https://github.com/silverstripe/silverstripe-framework/issues/8996

The TL:DR; is that we prefer protected if you think that there's a valid reason that someone might want to override this method (in the context of the purpose of the class), and you can use a docblock/attribute to mark as internal if we're worried about maintaining the protected method as public API in the eyes of SemVer.

I'm +1 for keeping this protected in this case.

Reply:
Yeah makes sense, an extension hook in a private had similarities with a protected in that there's API being added which you can't change within a major.

Probably with nothing though the API surface of the extension hook is a bit lower since it only covers the type of arguments passed to it, which usually matches the return type of the method it's contained in, rather than the entire method signature.

I think the intent of extension hooks is much clearer as in "here is the bit you that was designed to allow you to configure it" which a protected method by itself doesn't really communicate, especially given the proliferation of existing protected method in the codebase.

There's also the issue of composibility where multiple sources can add in their own extension implementations, while with a protected subclass you are limited to one implementation. This implementation is assumed to be a project. However this means that other Silverstripe modules that are composer required cannot add in their own implementation

Reply:
I think there's a valid use case for subclassing this class and it makes sense to keep the method protected. Maintenance burden looks pretty trivial as well.

I can see the general value of composability. But in this specific use case, what is being done is so simple and the results is so black and white, it's hard for me to image how two or more extensions could try to alter this behaviour and coexists.